dsls/sql/src/main/java/org/apache/beam/dsls/sql/interpreter/operator/functions/BeamSqlFunctionBinaryExpression.java

```diff
@@ -0,0 +1,61 @@
+package org.apache.beam.dsls.sql.interpreter.operator.functions;
+
+import java.util.List;
+import org.apache.beam.dsls.sql.interpreter.operator.BeamSqlExpression;
+import org.apache.beam.dsls.sql.interpreter.operator.BeamSqlPrimitive;
+import org.apache.beam.dsls.sql.schema.BeamSQLRow;
+import org.apache.calcite.sql.type.SqlTypeName;
+
+
+/**
+ * Base class for all binary functions.
+ */
+public abstract class BeamSqlFunctionBinaryExpression extends BeamSqlExpression {
+
+  private BeamSqlFunctionBinaryExpression(List<BeamSqlExpression> operands,
+      SqlTypeName outputType) {
+    super(operands, outputType);
+  }
+
+  public BeamSqlFunctionBinaryExpression(List<BeamSqlExpression> operands) {
+    this(operands, SqlTypeName.ANY);
+  }
+
+  @Override public boolean accept() {
+    boolean acceptance = false;
+    if (operands.size() != 2) {
+      return false;
+    }
+
+    if (SqlTypeName.NUMERIC_TYPES.contains(operands.get(0).getOutputType())
```

Comment:
there is some convenient methods in `BeamSqlExpression` you might find handy: `opType`, `op`, `opValueEvaluated`